can/bus.py

```diff
@@ -49,6 +57,7 @@ class BusABC(metaclass=ABCMeta):
     def __init__(
         self,
         channel: Any,
+        protocol: CanProtocol = CanProtocol.CAN_20,
```

Comment:
I'm not sure about this parameter here. I think subclasses should keep the parameters of its superclasses, which wouldn't make sense in this case.

What do you think about declaring `protocol` as a class attribute?

Reply:
Maybe as a protected instance attribute, i.e. (self._protocol)? If it was a class attribute, then it would have the same value for all instances, right?

Reply:
We could set the default value of `_protocol` as a class attribute. In the `__init__` method, the new value is assigned to the instance, not the class. So multiple instances will not share the same value.

Reply:
Not passing it in the super constructor but setting it directly sounds like a good plan. I assume you want to set it as a class attribute as a fallback in case it is not set in the subclass constructor? How would you be able to set it as an instance attribute if it is already present as a class attribute?

Reply:
```python3
class A:
    protocol = "Class Attribute"


class B(A):
    def __init__(self):
        self.protocol = "Instance Attribute"
        super().__init__()


print(A().protocol)
print(B().protocol)
```

This prints 
```text
Class Attribute
Instance Attribute
```

Reply:
Great, I wasn't aware that the way the attribute was accessed for assignment changed the assignment scope. I will change the implementation to follow this approach.

Reply:
@lumagi tell me if you need help with the changes

Reply:
Thanks for the offer. I have been a bit preoccupied with my time the last weeks, but I should be able to make some progress this week.

Reply:
The attribute should now be set inside the subclass constructor for all classes.